nifi-marklogic-processors/src/test/java/org/apache/nifi/marklogic/processor/QueryRowsMarkLogicIT.java

```diff
@@ -26,25 +28,76 @@ public void setup() {
 
         // Newline characters are making it difficult to assert on the entire response at once
-        final String response = new String(list.get(0).toByteArray());
+        final String response = new String(files.get(0).toByteArray());
         assertTrue(response.startsWith("Example.default.Id"));
         assertTrue(response.contains("0"));
         assertTrue(response.contains("1"));
         assertTrue(response.contains("2"));
         assertTrue(response.contains("3"));
         assertTrue(response.contains("4"));
+
+        assertEquals(0, runner.getFlowFilesForRelationship(QueryRowsMarkLogic.FAILURE).size());
+    }
+
+    @Test
+    public void testWithIncomingFlowFile() {
+        final String serializedPlan = "{\"$optic\":{\"ns\":\"op\", \"fn\":\"operators\", \"args\":[{\"ns\":\"op\", \"fn\":\"from-view\", \"args\":[\"Example\", \"default\", null, null]}, {\"ns\":\"op\", \"fn\":\"order-by\", \"args\":[[{\"ns\":\"op\", \"fn\":\"col\", \"args\":[\"Id\"]}]]}]}}";
+
+        TestRunner runner = getNewTestRunner(QueryRowsMarkLogic.class);
+        runner.setProperty(QueryRowsMarkLogic.PLAN, serializedPlan);
+
+        MockFlowFile mockFlowFile = new MockFlowFile(System.currentTimeMillis());
+        Map<String, String> attributes = new HashMap<>();
+        attributes.put("test-param1", "value1");
+        mockFlowFile.putAttributes(attributes);
+        runner.enqueue(mockFlowFile);
+
+        runner.run();
+
+        List<MockFlowFile> files = runner.getFlowFilesForRelationship(QueryRowsMarkLogic.ORIGINAL);
+        assertEquals(1, files.size(), "A new FlowFile should have been created and sent to Original");
+        MockFlowFile originalFile = files.get(0);
+        assertEquals("value1", originalFile.getAttribute("test-param1"), "Expecting the enqueued FlowFile to be sent " +
+            "to Original, and thus it should have the test-param1 attribute");
+        assertEquals(serializedPlan, originalFile.getAttribute("marklogic-optic-plan"), "The serialized plan should be" +
+            " on the success FlowFile so that a user knows what was executed to produce the rows");
+
+        files = runner.getFlowFilesForRelationship(QueryRowsMarkLogic.SUCCESS);
+        assertEquals(1, files.size());
+        MockFlowFile successFlowFile = files.get(0);
+        assertEquals("value1", successFlowFile.getAttribute("test-param1"), "The success FlowFile should have been " +
+            "created from the original FlowFile, so test-param1 should exist");
+
+        assertEquals(0, runner.getFlowFilesForRelationship(QueryRowsMarkLogic.FAILURE).size());
+    }
+
+    @Test
+    public void invalidPlan() {
+        final String serializedPlan = "invalid Plan";
+
+        TestRunner runner = getNewTestRunner(QueryRowsMarkLogic.class);
+        runner.setProperty(QueryRowsMarkLogic.PLAN, serializedPlan);
+
+        runner.run();
+
+        assertEquals(1, runner.getFlowFilesForRelationship(QueryRowsMarkLogic.FAILURE).size());
```

Comment:
Need 2 more assertions - make sure the count of FlowFiles in SUCCESS and ORIGINAL is zero. 

Then, get the FlowFile from the list sent to FAILURE and make sure that the `markLogicErrorMessage` attribute was set on it.